planner/core/logical_plans.go

```diff
@@ -268,6 +268,14 @@ type LogicalAggregation struct {
 	inputCount         float64 // inputCount is the input count of this plan.
 }
 
+// GetGroupByCols returns the groupByCols.
+func (la *LogicalAggregation) GetGroupByCols() []*expression.Column {
+	if la.groupByCols == nil {
+		la.collectGroupByColumns()
+	}
+	return la.groupByCols
```

Comment:
If `GroupByItems` changed, the `grouByCols` need to be recomputed. Should we need to consider this case here?

Reply:
If we changed the `GroupByItems`, we should explicitly call `collectGroupByColumns`. This method is just used for the new created Aggregations.

Reply:
> This method is just used for the new created Aggregations.

https://github.com/pingcap/tidb/blob/66a311c945dd153705a1071ed7ca086ee8671216/planner/core/rule_aggregation_push_down.go#L363-L375

Reply:
Yeah, so it explicitly calls the `collectGroupByColumns`. I mean `GetGroupByCols` is used for the new created Aggregations. Besides, we also use it to get `la.groupByCols` in cascades package.

Reply:
Ok，I misunderstanded that `this` refering to `collectGroupByColumns`. Anyway, I think it would be better to add some comment to avoid misusing `GetGroupByCols`.

Reply:
Sure, thanks ^.^